pkg/sidecar/sidecar.go

```diff
@@ -84,16 +89,48 @@ func Run(argv []string, version string) int {
 
 	klog.Infof("CSI driver name: %q", rt.DriverName)
 
-	// TBD May need to exposed metric HTTP end point
-	// here because the wait for the CSI driver is open ended.
+	rt.MetricsManager = metrics.NewMetricsManager()
+	wg := &sync.WaitGroup{}
+
+	mux := http.NewServeMux()
+	if *s.httpEndpoint != "" {
+		err := rt.MetricsManager.PrepareMetricsPath(mux, *s.metricsPath, promklog{})
+		if err != nil {
+			klog.Errorf("Failed to prepare metrics path: %s", err.Error())
+			os.Exit(1)
+		}
+		klog.Infof("Metrics path successfully registered at %s", *s.metricsPath)
+	}
 
 	grpcServer, err := startGRPCServerAndValidateCSIDriver(rt)
 	if err != nil {
 		klog.Error(err)
 		return 1
 	}
 
-	// TODO: Start the HTTP metrics server here.
+	// start listening & serving http endpoint if set
+	if *s.httpEndpoint != "" {
+		l, err := net.Listen("tcp", *s.httpEndpoint)
+		if err != nil {
+			klog.Fatalf("failed to listen on address[%s], error[%v]", *s.httpEndpoint, err)
+		}
+		srv := &http.Server{Addr: l.Addr().String(), Handler: mux}
+		go func() {
+			defer wg.Done()
+			if err := srv.Serve(l); err != http.ErrServerClosed {
+				klog.Fatalf("failed to start endpoint at:%s/%s, error: %v", *s.httpEndpoint, *s.metricsPath, err)
+			}
+		}()
+		klog.Infof("Metrics http server successfully started on %s, %s", *s.httpEndpoint, *s.metricsPath)
+		defer func() {
+			err := srv.Shutdown(context.Background())
+			if err != nil {
+				klog.Errorf("Failed to shutdown metrics server: %s", err.Error())
+			}
+			klog.Infof("Metrics server successfully shutdown")
+			wg.Done()
```

Comment:
Is it valid to call `wg.Done()` multiple times?

Reply:
No we shouldn't be doing this, updated the code to remove this.